jupyterlab_github/__init__.py

```diff
@@ -14,47 +14,56 @@
 
 
 link_regex = re.compile(r'<([^>]*)>;\s*rel="([\w]*)\"')
-GITHUB_API = 'https://api.github.com'
+
 
 class GitHubConfig(Configurable):
     """
-    A Configurable that declares the 'client_id' and 'client_secret'
-    parameters.
+    A Configurable that declares the `access_token` parameter.
     """
-    client_id = Unicode('', config=True,
-        help='The Client ID for the GitHub OAuth app')
-    client_secret = Unicode('', config=True,
-        help='The Client secret for the GitHub OAuth app')
+    api_url = Unicode(
+        'https://api.github.com', config=True,
+        help="The url for the GitHUb api"
+    )
+    access_token = Unicode(
+        '', config=True,
+        help="The GitHub access token"
+    )
+    validate_cert = Bool(
```

Comment:
Can you describe what this option is meant to do? I don't have a strong conception of it, or why it is now necessary.

Reply:
It stops SSL certificate validation for the requests to GitHub. Since Python libs don't in general use the Windows Certificate store they don't recognise the certificates used for our GitHub Enterprise so this just lets the request succeed by bypassing the validation.

Perhaps it should come with a warning that it's not a good idea in general.

Reply:
Sure, a warning makes sense. I'd just like good-practices to be documented, especially for things that I don't understand well myself.